onebusaway-gtfs-transformer/src/main/java/org/onebusaway/gtfs_transformer/impl/AnomalyCheckFutureTripCounts.java

```diff
@@ -0,0 +1,314 @@
+            if (dateTripMap.get(day) == null){
+                break;
+            }
+            int tripCount = dateTripMap.get(day);
+            if ((tripCount < dayAvgTripsMap.get(dayOfWeekFormat.format(day))*(1+percentageMatch/100)) &&
+                    (tripCount > dayAvgTripsMap.get(dayOfWeekFormat.format(day))*(1-percentageMatch/100))){
+                _log.info(day + " has " + tripCount + " trips, and that's within reasonable expections");
+                dayAvgTripsMapUpdate.get(dayOfWeekFormat.format(day)).add((double) tripCount);
+            }
+            else if (holidays.contains(day) & (tripCount < dayAvgTripsMap.get("Holiday")*(1+percentageMatch/100)
+                    && (tripCount > dayAvgTripsMap.get("Holiday")*(1-percentageMatch/100)))) {
+                _log.info(day + " has " + tripCount + " trips, is a holiday, and that's within reasonable expections");
+                dayAvgTripsMapUpdate.get("Holiday").add((double) tripCount);
+            }
+            else if (datesToIgnore.contains(day)) {
+                _log.info(day + " has " + tripCount + " trips, and we are ignoring this possible anomoly");
+            }
+            else {
+                _log.info(day + " has " + tripCount + " trips, this may indicate a problem.");
+                if (!silentMode) {
+                    es.publishMessage(getTopic(), day.toString() + " has " + tripCount + " trips, this may indicate a problem.");
+                }
+            }
+            dayCounter ++;
+        }
+
+
+        for (String day: days) {
+            try {
+                double tripsUpcoming = dayAvgTripsMapUpdate.get(day).stream().mapToDouble(a -> a).average().getAsDouble();
+                double currentAvg = dayAvgTripsMap.get(day);
+                int aprox_hours_per_month = 28 * 24;
+                double suggestedAvg = currentAvg + ((tripsUpcoming - currentAvg) / (aprox_hours_per_month));
+                _log.info(day + "," + suggestedAvg);
+            } catch (Exception e) {
+
+            }
+        }
+    }
+
+    private Map<Date, Integer> getDateTripMap(GtfsMutableRelationalDao dao){
+        Map<Date, Integer> dateTripMap = new HashMap<Date, Integer>();
+        for (Trip trip : dao.getAllTrips()) {
+            //_log.info(trip.toString());
+            //check for service
+            boolean hasCalDateException = false;
+            //are there calendar dates?
+            for (ServiceCalendarDate calDate : dao.getCalendarDatesForServiceId(trip.getServiceId())) {
+                //_log.info(calDate.toString());
+                Date date = constructDate(calDate.getDate());
+                if (dateTripMap.get(date) == null) {
+                    dateTripMap.put(date, 1);
+                } else {
+                    //Update this code so you are not using primitives and can just change object value rather than many puts depending on speed cost-- benchmark it!
```

Comment:
Is this a TODO around a suggested improvement?  I don't quite understand the comment.